h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -930,6 +933,8 @@ public void init(boolean expensive) {
       }
     }
     if (expensive) {
+      if (_parms._nparallelism == 0)
+        _parms._nparallelism = H2O.NUMCPUS;
```

Comment:
not going to do a full review, but first time I looked at this PR, I was concerned by how parallelism is handled in some recent algos:
we add a parallelism property for individual algo, and it's not clear how it combines with parallelism in CV for example, basically what do we call a `model` when we say we're training N models in parallel? and apparently it doesn't include the CV models, which I think can become a serious concern in some situations.
I wished model training parallelism was handled differently (more globally?) so that at any time we have a guarantee that we're not training more that N models in parallel.

Reply:
Yes, this is not optimal. H2O.NUMCPUS is almost never right.